src/components/VisualizationStep.tsx

```diff
@@ -1,35 +1,29 @@
+import { AppendStepButton } from './AppendStepButton';
+import { BranchBuilder } from './BranchBuilder';
+import { PrependStepButton } from './PrependStepButton';
+import './Visualization.css';
 import { MiniCatalog } from '@kaoto/components';
 import { usePosition } from '@kaoto/hooks';
 import { StepsService, VisualizationService } from '@kaoto/services';
-import {
-  useIntegrationJsonStore,
-  useNestedStepsStore,
-  useSettingsStore,
-  useVisualizationStore
-} from '@kaoto/store';
+import { useSettingsStore, useVisualizationStore } from '@kaoto/store';
 import { IStepProps, IVizStepNodeData } from '@kaoto/types';
 import { AlertVariant, Popover, Tooltip } from '@patternfly/react-core';
 import { CubesIcon, MinusIcon } from '@patternfly/react-icons';
 import { useAlert } from '@rhoas/app-services-ui-shared';
 import { Handle, NodeProps } from 'reactflow';
-import { AppendStepButton } from './AppendStepButton';
-import { BranchBuilder } from './BranchBuilder';
-import { PrependStepButton } from './PrependStepButton';
-import './Visualization.css';
 
 const currentDSL = useSettingsStore.getState().settings.dsl.name;
 
 // Custom Node type and component for React Flow
 const VisualizationStep = ({ data }: NodeProps<IVizStepNodeData>) => {
   const endStep = StepsService.isEndStep(data.step!);
-  const nestedStepsStore = useNestedStepsStore();
   const visualizationStore = useVisualizationStore();
-  const integrationJsonStore = useIntegrationJsonStore();
-  const visualizationService = new VisualizationService(integrationJsonStore, visualizationStore);
-  const stepsService = new StepsService(integrationJsonStore, nestedStepsStore, visualizationStore);
+  const visualizationService = new VisualizationService();
+  const stepsService = new StepsService();
   const showBranchesTab = VisualizationService.showBranchesTab(data.step);
   const showStepsTab = VisualizationService.showStepsTab(data);
   const supportsBranching = StepsService.supportsBranching(data.step);
+  const parentStepId = data.branchInfo?.parentStepUuid ?? undefined;
```

Comment:
[curious-question] shouldn't this be `undefined` if `branchInfo` is `undefined`? Maybe we don't need the `?? undefined` part

Reply:
fixned :+1: